### DISCUSSION TITLE
How to attach object viewer to the game

### DISCUSSION THREAD

**Discussion by FlameHorizon** [Albeoris/Memoria#192] [Q&A]
@Albeoris Can you please explain, how have you attached object viewer mentioned here (https://github.com/Albeoris/Memoria/issues/179#issuecomment-1016887414) to the game?

**Comment by Albeoris** [comment]
0. Make sure the launcher is running x64 version of the game.
1. Extract the latest release of [BepInEx 5 x64](https://github.com/BepInEx/BepInEx/releases/download/v5.4.19/BepInEx_x64_5.4.19.0.zip) to the game folder.
2. Extract the latest release of  [UnityExplorer (BepInEx5.Mono)](https://github.com/sinai-dev/UnityExplorer/releases/tag/4.5.6) to FF9\BepInEx\plugins
3. Run the game.

Also you can try [BepInEx 6 x64 (BepInEx Unity Mono for Windows x64 games)](https://builds.bepinex.dev/projects/bepinex_be) with UnityExplorer for BepInEx6.Mono

Please write if you succeeded or not.

**Reply by DV666** [reply]
I tried on my side but it doesn't seem to work :/

When I put BepInEx 5 or 6 in the "Final Fantasy IX" folder, nothing happens.
However, when I put it in the x64 folder, BepInEx seems to react but can't install itself.

With version 5, some files were created but nothing more. Can't launch the game, nothing happens.
With version 6, the game starts but gets stuck on a black screen: when I activate the BepInEx terminal, I get this:

```
[Message: Preloader] BepInEx 6.0.0-be.546 - Unity
[Message: Preloader] BLEEDING EDGE Build #546 from 1504d80c06a5c0449a879bc3aa90c62e071ca448 at master
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   : Preloader] Running under Unity v5.2.3.11896475
[Info   : Preloader] CLR runtime version: 3.0.40818.0
[Info   : Preloader] Supports SRE: True
[Message: Preloader] Preloader started
[Info   :AssemblyPatcher] Loaded 1 patcher type from [BepInEx.Preloader.Unity 6.0.0.546]
[Info   : Preloader] 1 patcher plugin loaded
[Info   : Preloader] 10 assemblies discovered
[Message:AssemblyPatcher] Executing 1 patch(es)
[Message: Preloader] Preloader finished
Setting breakpad minidump AppID = 377840
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198050968543 [API loaded no]
```

**Reply by Albeoris** [reply]
Please ask for help on BepInEx Discord server:
https://discord.gg/wVseDFfD

**Reply by DV666** [reply]
I ask them on the Discord, it's seems BepInEx is not compatible with FF9, explanations by one of the dev (Horse) :

> Pretty sure all FF games that are made in Unity have the game assemblies encrypted 
> I remember a few people reporting similar errors, the reason simply is that the game has encrypted assemblies
> You can try changing the entry point as per guides in https://docs.bepinex.dev/articles/user_guide/troubleshooting.html#unity-5-and-older but I doubt it will help here
> Also yeah, for UnityMono games you should be using BepInEx 5 for now

I tried with other entry points but the BepInEx have difficulties to read the .bin files from the game... sad :( 
This tool would have been so good to help my mod but Tirlititi showed me how to use the Memoria.Client.exe.

**Reply by Albeoris** [reply]
> I ask them on the Discord, it's seems BepInEx is not compatible with FF9, explanations by one of the dev (Horse) :
> 
> > Pretty sure all FF games that are made in Unity have the game assemblies encrypted
> > I remember a few people reporting similar errors, the reason simply is that the game has encrypted assemblies
> > You can try changing the entry point as per guides in https://docs.bepinex.dev/articles/user_guide/troubleshooting.html#unity-5-and-older but I doubt it will help here
> > Also yeah, for UnityMono games you should be using BepInEx 5 for now
> 
> I tried with other entry points but the BepInEx have difficulties to read the .bin files from the game... sad :( This tool would have been so good to help my mod but Tirlititi showed me how to use the Memoria.Client.exe.

Ok, it's not a problem, we just cannot use the BepInEx.
You can try UnityExplorer for MelonLoader.
If no success, we can use the Standalone UnityExplorer and load it in game via Assembly.Load

**Reply by Albeoris** [reply]
@DV666, good day, what the end of this story? :) I guess we can include auto-install some of mod loader or write a new one to run UnityExplorer in the game.

**Reply by DV666** [reply]
Hi @Albeoris! Really sorry for the response time... I don't use the “Discussions” tab very often ^^'
This story is quite old for me, and for my part, I gave up on this idea for FF9 (long time ago, I had tested for FF PR out of curiosity and had no trouble using for that.)
At the moment, I don't need BepInEx for my projects anymore... let's just say I've gained a bit of experience since then ^^ (well, even if I'm still far from having the level in C# hehe :p).

I'd also like to take this opportunity to leave you a message here, as I'd like to know whether me and @SamsamTS can have more rights to the Memoria repo? (to pull our own PRs).
For me, it's mainly for catalog updates, since Trance Seek is often updated. (For the rest, I always ask Tir for validation as i shared some ideas with him, like the SA extended system : https://github.com/Albeoris/Memoria/commit/54ca19629863d2c6512da37f674479e342afa268).
I sent you a message on Discord on April 7 but didn't get answer yet... I suppose you don't use Discord much (or you are flooded by MPs ^^)
I work a lot with @Tirlititi and @snouz and they know about this request too: do you think it would be possible?
I'll understand if you refuse :)

Thanks in advance for your answer !
